### DISCUSSION TITLE
Confused about clip vs. NDC space

### DISCUSSION THREAD

**Discussion by jwwalker** [google/filament#3820] [Q&A]
In the Filament Materials Guide, in several places you refer to "clip (NDC) space" or "normalized device (or clip) coordinate space", making it sound as if clip space and NDC were synonymous.  But in the definitions I've seen, clip space is 4D, NDC is 3D, and you get from clip to NDC by the perspective division.  Are you using different definitions?  If I had to guess, I's say that in places like the definition of `getClipFromViewMatrix()`, you mean clip space, but in the explanation of `vertexDomain : device`, you mean NDC.

**Comment by romainguy** [comment]
You are right, the documentation mixes clip and NDC and that's confusing. We need to fix this.

**Reply by romainguy** [reply]
@pixelflinger @bejado @prideout

**Reply by jwwalker** [reply]
Can you verify my guess that if the material specifies `vertexDomain : device`, then the vertices should be specified in NDC?  And then they will not be affected by any `clipSpaceTransform` that the vertex shader might specify?
